src/deluge/processing/sound/sound_instrument.cpp

```diff
@@ -395,20 +388,21 @@ void SoundInstrument::sendNote(ModelStackWithThreeMainThings* modelStack, bool i
 
 		arpeggiator.noteOff(arpSettings, noteCode, &instruction);
 
-		if (instruction.noteCodeOffPostArp != ARP_NOTE_NONE) {
-
+		for (int32_t n = 0; n < ARP_MAX_INSTRUCTION_NOTES; n++) {
+			if (instruction.noteCodeOffPostArp[n] != ARP_NOTE_NONE) {
```

Comment:
```suggestion
			if (instruction.noteCodeOffPostArp[n] == ARP_NOTE_NONE) { break; }
```

Should avoid re-indenting everything at least.